homework/list.js

```diff
@@ -1,44 +1,36 @@
 'use strict';
 
-(function() {
+(function(exports) {
 
+var ListManager = function() {
   var _listNoteContent = [];
   var _wrapper = document.querySelector('#note-list-wrapper');
+};
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
-  }
-
-  function onNoteOpen(event) {
+ListManager.prototype = {  
+  onNoteOpen(event) {
     if (event.target.classList.contains('note-title')) {
       var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
+      var content = this._listNoteContent[id];
       window.dispatchEvent(new CustomEvent('note-open',
         { detail: content }));
     };
-  }
-
-  function preloadFirstNote() {
-    if (_listNoteContent.length !== 0) {
-      var content = _listNoteContent[0];
+  },
+ 
+   preloadFirstNote() {
+    if (this_listNoteContent.length !== 0) {
```

Comment:
If a function body is wrapped by an `if...else` with only one branch, moving the condition into the caller rather than caller is recommended.